irc_client/backend.py

```diff
@@ -165,6 +165,16 @@ def __init__(self, server_config: config.ServerConfig):
         # the replies are collected here before emitting a self_joined event
         self._names_replys: dict[str, list[str]] = {}  # {channel: [nick1, nick2, ...]}
 
+    def get_current_config(self, channels: list[str]) -> config.ServerConfig:
```

Comment:
Possible bug: channel list becomes empty if you close the program before it fully starts and joins the channels.

The proper fix would be to replace `self._autojoin` with a list of channels where the user actually is.